patches/chromium/fix_resolve_dynamic_background_material_update_issue_on_windows_11.patch

```diff
@@ -0,0 +1,85 @@
+@@ -94,6 +94,8 @@ class VIEWS_EXPORT DesktopWindowTreeHostWin
+   // false.
+   void FinishTouchDrag(gfx::Point screen_point);
+ 
++  void SetIsTranslucent(bool is_translucent);
++
+  protected:
+   // Overridden from DesktopWindowTreeHost:
+   void Init(const Widget::InitParams& params) override;
+diff --git a/ui/views/win/hwnd_message_handler.cc b/ui/views/win/hwnd_message_handler.cc
+index 09509acd13511525b0eb7b9afa972e20d8d3601c..e0024f400de5c2de60524b4b75047062cde2f31f 100644
+--- a/ui/views/win/hwnd_message_handler.cc
++++ b/ui/views/win/hwnd_message_handler.cc
+@@ -1037,7 +1037,14 @@ void HWNDMessageHandler::SizeConstraintsChanged() {
+   //   allowing ui::GetResizableFrameThickness() to be used consistently when
+   //   removing the visible system frame.
+   const bool had_caption_on_init = window_style() & WS_CAPTION;
+-  const bool can_resize = !is_translucent_ && delegate_->CanResize();
++
++  // In Chromium, the !is_translucent_ check was introduced for Glic-specific
++  // behavior. Since Electron does not use Glic, this restriction can be safely
++  // removed. Keeping the is_translucent_ check disables maximization for
++  // translucent framed windows. Original code: !is_translucent_ &&
++  // delegate_->CanResize() See:
++  // https://chromium-review.googlesource.com/c/chromium/src/+/6372329
++  const bool can_resize = delegate_->CanResize();
+   const bool can_maximize = can_resize && delegate_->CanMaximize();
+ 
+   auto set_style_func = [&style](LONG bit, bool should_set) {
+@@ -1629,11 +1636,16 @@ void HWNDMessageHandler::ResetWindowRegion(bool force, bool redraw) {
+   // through, but that isn't the case when using Direct3D to draw transparent
+   // windows. So we route translucent windows throught to the delegate to
+   // allow for a custom hit mask.
+-  if (!is_translucent_ && !custom_window_region_.is_valid() &&
++  // patch: fix_resolve_dynamic_background_material_update_issue_on_windows_11
++  // Our translucent windows use the native frame by default, and we should not
++  // set a custom region when the window is maximized; otherwise, it will cause
++  // a white title bar to appear under Windows 11.
++  if (!custom_window_region_.is_valid() &&
+       (IsFrameSystemDrawn() || !delegate_->HasNonClientView())) {
+     if (force) {
+       SetWindowRgn(hwnd(), nullptr, redraw);
+     }
++
+     return;
+   }
+ 
+@@ -2310,6 +2322,12 @@ void HWNDMessageHandler::OnMoving(UINT param, const RECT* new_bounds) {
+ LRESULT HWNDMessageHandler::OnNCActivate(UINT message,
+                                          WPARAM w_param,
+                                          LPARAM l_param) {
++  // patch: fix_resolve_dynamic_background_material_update_issue_on_windows_11
++  // When in fullscreen or when thick_frame_ is false, we should skip the update
++  // to avoid showing a white title bar.
++  if (IsFullscreen() || !(GetWindowLong(hwnd(), GWL_STYLE) & WS_THICKFRAME)) {
```

Comment:
In theory, when in fullscreen mode, there is no non-client area, so the processing can be skipped to avoid background material flickering and incorrect titlebars on multi-monitor setups (#46882). When thickFrame is false in Electron, the fullscreen check becomes unreliable, so window styles are used for detection instead.